pkg/bot/discord.go

```diff
@@ -0,0 +1,133 @@
+package bot
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/infracloudio/botkube/pkg/config"
+	"github.com/infracloudio/botkube/pkg/execute"
+	"github.com/infracloudio/botkube/pkg/logging"
+)
+
+// DiscordBot listens for user's message, execute commands and sends back the response
+type DiscordBot struct {
+	Token          string
+	AllowKubectl   bool
+	RestrictAccess bool
+	ClusterName    string
+	ChannelID      string
+	BotID          string
+}
+
+// discordMessage contains message details to execute command and send back the result
+type discordMessage struct {
+	Event         *discordgo.MessageCreate
+	BotID         string
+	Request       string
+	Response      string
+	IsAuthChannel bool
+	Session       *discordgo.Session
+}
+
+// NewDiscordBot returns new Bot object
+func NewDiscordBot() Bot {
+	c, err := config.New()
+	if err != nil {
+		logging.Logger.Fatal(fmt.Sprintf("Error in loading configuration. Error:%s", err.Error()))
+	}
+	return &DiscordBot{
+		Token:          c.Communications.Discord.Token,
+		BotID:          c.Communications.Discord.BotID,
+		AllowKubectl:   c.Settings.AllowKubectl,
+		RestrictAccess: c.Settings.RestrictAccess,
+		ClusterName:    c.Settings.ClusterName,
+		ChannelID:      c.Communications.Discord.Channel,
+	}
+}
+
+// Start starts the slacknot RTM connection and listens for messages
+func (b *DiscordBot) Start() {
+	// var botID string
+	api, err := discordgo.New("Bot " + b.Token)
+	if err != nil {
+		logging.Logger.Error("error creating Discord session,", err)
+		return
+	}
+
+	// Register the messageCreate func as a callback for MessageCreate events.
+	api.AddHandler(func(s *discordgo.Session, m *discordgo.MessageCreate) {
+		dm := discordMessage{
+			Event:   m,
+			BotID:   b.BotID,
+			Session: s,
+		}
+
+		dm.HandleMessage(b)
+	})
+
+	// Open a websocket connection to Discord and begin listening.
+	go func() {
+		err := api.Open()
+		if err != nil {
+			logging.Logger.Error("error opening connection,", err)
+			return
+		}
+	}()
+
+	// Wait here until CTRL-C or other term signal is received.
+	logging.Logger.Info("Bot is now running.")
```

Comment:
It the comment relevant here? Also please have uniform Bot start log message just like we have for other mediums